tests/McpEndToEndTest/McpPromptsClasses.cs

```diff
@@ -0,0 +1,92 @@
+            {
+                new PromptMessage("Create a Person object with Name: 'Alice', Surname: 'Smith', Age: '25', Address: { Street: '456 Elm St', City: 'Springfield', PostalCode: '67890', Country: 'USA' }. Then call ProcessPerson with this object and return the response.")
+            };
+        }
+
+        [McpServerPrompt("processperson_workflow", "Demonstrate a workflow using ProcessPerson tool")]
+        public static PromptMessage[] ProcessPersonWorkflow()
+        {
+            return new PromptMessage[]
+            {
+                new PromptMessage("Call GetDefaultPerson, then pass its output into ProcessPerson. Return both the initial and processed person object.")
+            };
+        }
+
+        [McpServerPrompt("get_default_address_integration", "Demonstrate integration with GetDefaultAddress tool")]
+        public static PromptMessage[] GetDefaultAddressIntegration()
+        {
+            return new PromptMessage[]
+            {
+                new PromptMessage("First call GetDefaultPerson, then call GetDefaultAddress. Combine both into a summary like: 'Person X lives at Y'.")
+            };
+        }
+
+        [McpServerPrompt("high_level_agent_prompt", "Adds logic and semantic intelligence on top of tools")]
+        public static PromptMessage[] HighLevelAgentPrompt()
+        {
+            return new PromptMessage[]
+            {
+                new PromptMessage("You're a data-summary agent. Fetch the default person and address, then produce a human -readable summary. If age > 30, add '(senior)', otherwise '(junior)' at the end.")
+            };
+        }
+
+        [McpServerPrompt("confirmation_flow", "Tests conversational context, user confirmation, and conditional logic")]
+        public static PromptMessage[] ConfirmationFlow()
+        {
+            return new PromptMessage[]
+            {
+                new PromptMessage("Before calling ProcessPerson, ask: 'Do you want to process the default person? [yes/no]'. If user says yes, proceed; else return 'Operation canceled.'")
+            };
+        }
+
+        [McpServerPrompt("summarize_person", "Fetches person and address, processes the person, uses ageThreshold to label as junior or senior.")]
+        [McpPromptParameter("ageThreshold", "The age threshold to determine if the person is a senior or junior.")]
+        public static PromptMessage[] SummarizePerson(string ageThreshold)
+        {
+            return new PromptMessage[]
+            {
+                new PromptMessage($"Please perform the following steps:\r\n1. Call GetDefaultPerson() -> person.\r\n2. Call GetDefaultAddress() -> address.\r\n3. If person.Age > {ageThreshold} then set label = \"senior\"; otherwise set label = \"junior\".\r\n4. Call ProcessPerson(person) -> processed.\r\n5. Return a JSON object:\r\n{{ \"name\": person.Name, \"age\": person.Age, \"label\": label, \"address\": address, \"processed\": processed }}")
```

Comment:
everything needs to be string encoded. so, this will break. This will actually work (or manual encoding):


```suggestion
                new PromptMessage(JsonSerializer.SerializeObject($"Please perform the following steps:\r\n1. Call GetDefaultPerson() → person.\r\n2. Call GetDefaultAddress() → address.\r\n3. If person.Age > {ageThreshold ?? string.Empty} then set label = \"senior\"; otherwise set label = \"junior\".\r\n4. Call ProcessPerson(person) → processed.\r\n5. Return a JSON object:\r\n{{ \"name\": person.Name, \"age\": person.Age, \"label\": label, \"address\": address, \"processed\": processed }}").TrimStart('"').TrimEnd('"'))
```

Reply:
For performance, I would prefer a manual encoded string, so there is no need to call the Json serializer.